pbs/buckets/cpm_bucketmanager.go

```diff
@@ -110,30 +110,45 @@ func getCpmStringValue(cpm float64, config priceBucketConf, precision int) strin
 		}
 	}
 	if increment > 0 {
-		cpmStr = getCpmTarget(cpm, increment, precision)
+		cpmStr = getCpmTarget(cpm, increment, precision, granularityMultipler)
 	}
 	return cpmStr
 }
 
-func getCpmTarget(cpm float64, increment float64, precision int) string {
+func getCpmTarget(cpm float64, increment float64, precision int, granularityMultipler float64) string {
+	var cpmTarget string
 	// Probably don't need this default check given it is in getCpmStringValue
 	if precision == 0 {
 		precision = DEFAULT_PRECISION
 	}
 	roundedCPM := math.Floor(cpm/increment) * increment
-	return strconv.FormatFloat(roundedCPM, 'f', precision, 64)
+	cpmBeforeMultiplier := strconv.FormatFloat(roundedCPM, 'f', precision, 64)
+
+	if granularityMultipler != 0 {
+		cpmBeforeMultiplierF, err := strconv.ParseFloat(cpmBeforeMultiplier, 64)
+		if err != nil {
+			cpmTarget = cpmBeforeMultiplier
+		} else {
+			cpmWithMultiplier := cpmBeforeMultiplierF * granularityMultipler
```

Comment:
couldn't think of a more efficient way to multiply cpm with granularity multiplier since cpm needs to be rounded before multiplication happens.